alchemiscale/compute/service.py

```diff
@@ -178,62 +234,94 @@ def execute(self, task: ScopedKey) -> ScopedKey:
 
         # execute the task; this looks the same whether the ProtocolDAG is a
         # success or failure
+        shared = self.shared_basedir / str(protocoldag.key) / str(self.counter)
+        shared.mkdir(parents=True)
+
         protocoldagresult = execute_DAG(
             protocoldag,
-            shared=self.shared,
+            shared=shared,
+            scratch_basedir=self.scratch_basedir,
+            keep_scratch=self.keep_scratch,
+            raise_error=False
         )
 
         # push the result (or failure) back to the compute API
         result_sk = self.push_result(task, protocoldagresult)
 
         return result_sk
 
+    def cycle(self, task_limit):
+        if task_limit is not None:
+            if self.counter >= task_limit:
+                self.logger.info("Performed %s tasks; beyond task limit %s", self.counter, task_limit)
+                return
+
+        # claim tasks from the compute API
+        self.logger.info("Claiming tasks")
+        tasks: List[ScopedKey] = self.claim_tasks(self.limit)
+        self.logger.info("Claimed %d tasks", len([t for t in tasks if t is not None]))
```

Comment:
I would also dump the scoped keys to the log?

Reply:
We already indicate which task we execute and finish further below. I think the log may be too noisy doing it here as well.